code/controllers/subsystem/vote.dm

```diff
@@ -5,437 +5,16 @@ SUBSYSTEM_DEF(vote)
-		if(admin)
-			dat += "<a href='?src=[UID()];vote=custom'>Custom</a></li>"
-		dat += "</ul></div><hr>"
-	var/datum/browser/popup = new(C.mob, "vote", "Voting Panel", nref=src)
-	popup.set_content(dat)
-	popup.open()
-
-/datum/controller/subsystem/vote/proc/update_panel(client/C)
-	C << output(url_encode(vote_html(C)), "vote.browser:update_vote_div")
-
-/datum/controller/subsystem/vote/proc/vote_html(client/C)
-	. = ""
-	if(question)
-		. += "<h2>Vote: '[question]'</h2>"
-	else
-		. += "<h2>Vote: [capitalize(mode)]</h2>"
-	. += "Time Left: [time_remaining] s<hr><ul>"
-	for(var/i = 1, i <= choices.len, i++)
-		var/votes = choices[choices[i]]
-		if(!votes)
-			votes = 0
-		var/vote_count = null
-		if(check_rights(R_ADMIN, FALSE, C.mob))
-			vote_count = " ([votes] vote\s)"
-		if(current_votes[C.ckey] == i)
-			. += "<li><b><a href='?src=[UID()];vote=[i]'>[choices[i]][vote_count]</a></b></li>"
-		else
-			. += "<li><a href='?src=[UID()];vote=[i]'>[choices[i]][vote_count]</a></li>"
-
-	. += "</ul>"
-
-
-/datum/controller/subsystem/vote/Topic(href,href_list[],hsrc)
-	if(!usr || !usr.client)
-		return	//not necessary but meh...just in-case somebody does something stupid
-	var/admin = check_rights(R_ADMIN,0)
-	if(href_list["close"])
-		voting -= usr.client
-		return
-	switch(href_list["vote"])
-		if("open")
-			// vote proc will automatically get called after this switch ends
-		if("cancel")
-			if(admin && mode)
-				var/votedesc = capitalize(mode)
-				if(mode == "custom")
-					votedesc += " ([question])"
-				log_and_message_admins("cancelled the running '[votedesc]' vote.")
-				reset()
-		if("toggle_restart")
-			if(admin)
-				GLOB.configuration.vote.allow_restart_votes = !GLOB.configuration.vote.allow_restart_votes
-				log_and_message_admins("has [GLOB.configuration.vote.allow_restart_votes ? "enabled" : "disabled"] public restart voting.")
-		if("toggle_gamemode")
-			if(admin)
-				GLOB.configuration.vote.allow_mode_votes = !GLOB.configuration.vote.allow_mode_votes
-				log_and_message_admins("has [GLOB.configuration.vote.allow_mode_votes ? "enabled" : "disabled"] public gamemode voting.")
-		if("restart")
-			if(GLOB.configuration.vote.allow_restart_votes || admin)
-				initiate_vote("restart",usr.key)
-		if("gamemode")
-			if(GLOB.configuration.vote.allow_mode_votes || admin)
-				initiate_vote("gamemode",usr.key)
-		if("map")
-			if(admin)
-				initiate_vote("map", usr.key)
-		if("crew_transfer")
-			if(GLOB.configuration.vote.allow_restart_votes || admin)
-				initiate_vote("crew transfer", usr.key)
-		if("custom")
-			if(admin)
-				initiate_vote("custom",usr.key)
-		else
-			submit_vote(usr.ckey, round(text2num(href_list["vote"])))
-			update_panel(usr.client)
-			return
-	usr.vote()
-
-
-/mob/verb/vote()
-	set category = "OOC"
-	set name = "Vote"
+	new /datum/vote/crew_transfer
 
-	if(SSvote)
-		SSvote.browse_to(client)
+/datum/controller/subsystem/vote/Topic(href, list/href_list)
+	if(href_list["vote"] == "open")
+		usr.client.vote()
```

Comment:
Part of me feels like this should just be thrown on `client/Topic`

I'll let review decide.